client/state/index.js

```diff
@@ -149,13 +149,17 @@ export function createReduxStore( initialState = {} ) {
 
 	const middlewares = [
 		thunkMiddleware,
+		// We need the data layer middleware to be used as early
+		// as possible, before any side effects.
+		// This avoids, for example, side effect actions like analytics
+		// being dispatched more than once.
+		require( './data-layer/wpcom-api-middleware.js' ).default,
```

Comment:
> test testing that the code we wrote is the code we wrote

That wasn't my intent with the above description, but I do agree that the test implementation is trivial and fragile. Mainly I want an extra warning if someone tries to add a new middleware. Not all folks read the comments.

As for your suggestion we can certainly do something like that, though test setup might be interesting. Perhaps we add a test in state/index, where we dispatch a network request and spy on each middleware?